CoreScriptsRoot/CoreScripts/ContextActionTouch.lua

```diff
@@ -171,6 +208,65 @@ function createNewButton(actionName, functionInfoTable)
 	return contextButton
 end
 
+function getButtonPosition(index, buttonSize)
+	local complete = false
+	local position, occupyingPosition
+
+	while not complete do
+		if buttonPositionsOccupied[index] then
+			index = index + 1
+		else
+			local info = buttonPositionTable[index]
+
+			-- Find the rotation and distance relative to jump button
+			local rotation, distance
+
+			if info ~= nil then
+				-- Use preset position
+				rotation, distance = unpack(info)
+			else
+				-- Auto generate position based on preset arcs
+				local arc = 0
+				local positionOnArc = 0
+				local totalInArc = 0
+				local indexOnArcs = index - #buttonPositionTable - 1
+				
+				local total = 0
+				for arcIndex=1, #buttonArcs do
+					local numInArc = buttonArcs[arcIndex]
+					local prevTotal = total
+					total = total + numInArc
+
+					if total > indexOnArcs then
+						arc = arcIndex
+						positionOnArc = indexOnArcs - prevTotal
+						totalInArc = numInArc
+						break
+					end
+				end
```

Comment:
The above arc calculation can be moved into its own function as well for increased readability.
Then we can replace the break with a return statement.
Also, if the number of buttons exceed all that can fit in the arcs does it always return position as 0?